pkg/utils/utils.go

```diff
@@ -624,3 +624,10 @@ func String(s *string) string {
 	}
 	return ""
 }
+
+func GetUriEncoded(uri string) string {
+	if uriSplit := strings.SplitN(uri, "?", 2); len(uriSplit) == 2 {
+		return uriSplit[0] + "?" + url.QueryEscape(uriSplit[1])
```

Comment:
shouldnt we PathEscape the first part of URL?

Reply:
shouldn't be required as we are making these calls to controller and the paths are known